### PR TITLE
feat(vh.py): add topic_ontology method to VirtualHavruta class

### PR DESCRIPTION
**Pull request by edemirci-aai**
- Implemented topic_ontology method to process extraction strings and retrieve topic descriptions from the Sefaria API.
- Added internal helper functions for fetching and processing topic data.
- Included logging statements to track the process and results.
- Implemented error handling to manage API call failures and ensure robustness.
- The method returns a dictionary with topic names as keys and their English descriptions as values.

### CODE REVIEW

File: VirtualHavruta/vh.py [hunk 1]
```diff
@@ -800,3 +801,73 @@ def merge_linker_refs(self, retrieved_docs: list, p_sorted_src_rel_dict: dict, p
         self.logger.info(f"MsgID={msg_id}. [FINAL LINKER REFERENCE MERGE OUTPUT] \n----p_sorted_src_rel_dict: {p_sorted_src_rel_dict} \n----p_src_data_dict: {p_src_data_dict} \n----p_src_ref_dict: {p_src_ref_dict}")
 
         return p_sorted_src_rel_dict, p_src_data_dict, p_src_ref_dict
+        
+    def topic_ontology(self, extraction: str = '', msgid: str = ''):
+        self.logger.info(f"MsgID={msgid}. Starting topic ontology process.")
+
+        def get_all_topics():
```
Comment (Paul-Yu-Chun-Chang) [inline VirtualHavruta/vh.py:808]: There's a need for a mechanism to regularly update all topics.json

File: VirtualHavruta/vh.py [hunk 2]
```diff
@@ -800,3 +801,73 @@ def merge_linker_refs(self, retrieved_docs: list, p_sorted_src_rel_dict: dict, p
...
+        self.logger.info(f"MsgID={msgid}. Extracted topic names: {topic_names}")
+
+        # Get all topics
+        all_topics = get_all_topics()
+
+        # Find slugs for the topic names
+        topic_slugs = find_topic_slugs(topic_names, all_topics)
+
+        # Get descriptions for the topic slugs
+        descriptions = get_topic_descriptions(topic_slugs)
```
Comment (Paul-Yu-Chun-Chang) [inline VirtualHavruta/vh.py:882]: Let's consider if there's still other valuable information from this API

File: VirtualHavruta/vh.py [hunk 3]
```diff
@@ -800,3 +801,73 @@ def merge_linker_refs(self, retrieved_docs: list, p_sorted_src_rel_dict: dict, p
...
+
+        # Create a dictionary with topic names as keys and their descriptions as values
+        final_descriptions = {}
+        for name, slug in zip(topic_names, topic_slugs):
+            if slug in descriptions:
+                final_descriptions[name] = descriptions[slug]
+
+        self.logger.info(f"MsgID={msgid}. Final topic descriptions: {final_descriptions}")
+        return final_descriptions
```
Comment (Paul-Yu-Chun-Chang) [inline VirtualHavruta/vh.py:891]: Please update the dev app workflow.

File: VirtualHavruta/vh.py (hunk 1 above)
Comment (edemirci-aai) [reply VirtualHavruta/vh.py:808]: done its in vh_topic.py in our uca instance

File: VirtualHavruta/vh.py (hunk 3 above)
Comment (edemirci-aai) [reply VirtualHavruta/vh.py:891]: updated slack_app_ontology.py in uca instance with this code

**Review by Paul-Yu-Chun-Chang — APPROVED** [review]

**Pull request merged**